casanovo/denovo/model_runner.py

```diff
@@ -163,18 +163,35 @@ def log_metrics(self, test_index: AnnotatedSpectrumIndex) -> None:
             Index containing the annotated spectra used to generate model
             predictions
         """
-        model_output = [psm.sequence for psm in self.writer.psms]
-        spectrum_annotations = [
-            test_index[i][4] for i in range(test_index.n_spectra)
-        ]
+        model_output = []
+        spectrum_annotations = []
+        psms = iter(self.writer.psms)
+        curr_psm = next(psms, None)
+
+        with test_index as t_ind:
+            for i in range(t_ind.n_spectra):
+                if curr_psm is None:
+                    break
+
+                spectrum_annotations.append(t_ind[i][4])
+                if curr_psm.spectrum_id == t_ind.get_spectrum_id(i):
+                    model_output.append(curr_psm.sequence)
+                    curr_psm = next(psms, None)
+                else:
+                    model_output.append("")
```

Comment:
**question**: Is this the correct behavior for handling skipped spectra? I'm not sure if skipped spectra should be ignored entirely or yield a peptide precision of 0.

Reply:
I think they should be treated as incorrect predictions with scores of -1.

Reply:
I agree. `None` might be a better value instead of an empty string.

Reply:
Using `None` causes `aa_match_batch` to throw an exception. Should `aa_match_batch` be updated to handle `None` or should we just go with the empty string?

Reply:
I'd suggest to update `aa_match_batch` to handle `None`, so that we have a clear semantic indication of the spectrum being skipped.